DreamJournal/DreamJournal/NewEntryView.swift

```diff
@@ -0,0 +1,206 @@
+//
+//  ContentView.swift
+//  DreamJournal
+//
+//  Created by CUBS Customer on 10/7/22.
+//
+
+import SwiftUI
+import CoreData
+
+struct CustomColor {
+    static let Navy = Color("Navy")
+    static let SkyPurple = Color("SkyPurple")
+    static let LavenderBox = Color("LavenderBox")
+}
+
+enum Category: String, Identifiable, CaseIterable {
+    var id: UUID {
+        return UUID()
+    }
+    
+    case nightmare = "Nightmare"
+    case neutral = "Neutral Dream"
+    case good = "Good Dream"
+}
+
+extension Category {
+    var title: String {
+        switch self {
+            case .nightmare:
+                return rawValue
+            case .neutral:
+                return rawValue
+            case .good:
+                return rawValue
+        }
+    }
+}
+
+struct NewEntryView: View {
+    init() {
+        UISegmentedControl.appearance().selectedSegmentTintColor = .orange
+//        UISegmentedControl.appearance().layer.backgroundColor = .blue
+    }
+    
+    
+    // bool for button to save entry
+    @State private var goToHome: Bool = false
+    @State private var title: String = ""
+    @State private var topic: String = ""
+    @State private var selectedCategory: Category = .neutral
+    @Environment(\.managedObjectContext) private var viewContext
+    
+    @FetchRequest(entity: Entry.entity(), sortDescriptors: [NSSortDescriptor(key: "date", ascending: false)]) private var allEntries: FetchedResults<Entry>
+
+       private func saveDream() {
+
+           do {
+               let entry = Entry(context: viewContext)
+               entry.title = title
+               entry.emotion = selectedCategory.rawValue
+               entry.date = Date()
+               entry.topic = topic
+               try viewContext.save()
+           } catch {
+               print(error.localizedDescription)
+           }
+
+       }
+    
+    private func styleForPriority(_ value: String) -> Color {
```

Comment:
You could put this as a var on `Category` itself if you want to. Feels a little cleaner to me to define in there.

Reply:
Definitely seems a bit cleaner-- I'll change this!